fix(symbolon): KeyringCredentialProvider is pub(crate) but re-exported pub

## Bug

`cargo check --workspace --all-features` fails because `KeyringCredentialProvider` is declared `pub(crate)` but re-exported as `pub` from `crates/symbolon/src/credential/mod.rs`.

```
error[E0365]: `KeyringCredentialProvider` is only public within the crate, and cannot be re-exported outside
  --> crates/symbolon/src/credential/mod.rs:19:9
```

Introduced by #2225 (full audit lint fix).

## Fix

The type is used from `crates/aletheia/src/commands/credential.rs` and `crates/aletheia/src/runtime.rs` — it must be `pub`. Change `pub(crate) struct KeyringCredentialProvider` → `pub struct KeyringCredentialProvider` and the `new()` ctor similarly.

## Impact

Blocks `cargo check --workspace --all-features` and any refactor that requires that gate (e.g., #2800 env::var consolidation).

--- a/crates/symbolon/src/credential/keyring_provider.rs
+++ b/crates/symbolon/src/credential/keyring_provider.rs
@@ -13,7 +13,7 @@ const DEFAULT_USERNAME: &str = "api-token";
 ///
 /// Falls through silently when the keyring is unavailable (headless server,
 /// no D-Bus session, locked keychain) so downstream providers get a chance.
-pub(crate) struct KeyringCredentialProvider {
+pub struct KeyringCredentialProvider {
     service: String,
     username: String,
 }
@@ -22,7 +22,7 @@ impl KeyringCredentialProvider {
     /// Create a provider using the default service name (`aletheia`) and
     /// username (`api-token`).
     #[must_use]
-    pub(crate) fn new() -> Self {
+    pub fn new() -> Self {
         Self {
             service: DEFAULT_SERVICE.to_owned(),
             username: DEFAULT_USERNAME.to_owned(),

